src/app/domain/backup/services/manageRemoteBackupConfig.ts

```diff
@@ -233,10 +238,25 @@ export const createRemoteBackupFolder = async (
   return remoteBackupConfig
 }
 
+const isFileCorrespondingToMedia = (file: File, media: Media): boolean => {
+  return (
+    file.name === media.name &&
+    new Date(file.created_at).getTime() === media.creationDate &&
+    new Date(file.updated_at).getTime() === media.modificationDate
+  )
+}
+
 const formatBackupedMedia = (
+  allMedias: Media[],
   deviceRemoteBackupConfig: RemoteBackupConfig,
   file: File
-): BackupedMedia => {
+): BackupedMedia | undefined => {
+  const correspondingMedia = allMedias.find(media =>
+    isFileCorrespondingToMedia(file, media)
+  )
```

Comment:
What is the perf cost here? I remember that we already have a lot of loops, we're adding one?

Reply:
My point of view of this for the moment (we are indeed adding a loop) : 
- the loop cost is okay
- the line `const allMedias = await getAllMedias(client)` is less okay
- we should be able to add some cache to this big request to the native side
- it only happens when _restoring a backup_ so I propose to do nothing for the moment

But I am of course open to everything.

Reply:
I don't know yet how to optimize that, but let's ship that way and once we know that it works, let's see how we can improve it if needed.